src/pip/_internal/pyproject.py

```diff
@@ -174,4 +174,9 @@ def load_pyproject_toml(
         backend = "setuptools.build_meta:__legacy__"
         check = ["setuptools>=40.8.0", "wheel"]
 
+    if backend == "setuptools.build_meta:__legacy__" and not has_setup:
```

Comment:
setup.py-less projects should be permitted if they specify a pyproject.toml

Reply:
In which case they don't use the legacy backend, or ?

Reply:
Or what if a project lists `setuptools.build_meta:__legacy__` as their build-backend? setup.py-less builds should definitely work then

Reply:
Then they get the behaviour from this PR. Why is that a problem?

Reply:
Right but `setuptools.build_meta:__legacy__` automatically generates a `setup.py` if it's missing

Reply:
https://github.com/pypa/setuptools/blob/main/setuptools/build_meta.py#L116

Reply:
I'm aware of the behaviour of setuptools, and that isn't an answer to the question asked.

Let me rephrase: Why is should "setup.py-less" projects work when they're using the `setuptools.build_meta:__legacy__` backend (whether implicitly, or through an explicit specification in the `pyproject.toml` file)?

Reply:
Note that there's a very straightforward solution in those cases -- add the right `pyproject.toml` content.